### PR TITLE
Check that datamodel can target WASM in CI 

### PR DESCRIPTION
**Pull request by tomhoule**
See commit messages for details. TL;DR it's a nice property, let's make sure we don't introduce regressions.

### CODE REVIEW

**Review by pimeys — APPROVED** [review]
Wow. This was easier than I expected. Greate, sashay away!

File: .github/workflows/build-wasm.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    branches:
+      - master
+  pull_request:
+jobs:
+  build:
+    name: "WebAssembly build ${{ github.event.ref }} for commit ${{ github.event.inputs.commit }}"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Output link to real commit
+        run: echo ${{ github.repository }}/commit/${{ github.event.inputs.commit }}
```
Comment (tomhoule) [inline .github/workflows/build-wasm.yml:13]: Is this necessary? I picked that up from the windows build, we may not need it.

File: .github/workflows/build-wasm.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+      - name: Checkout ${{ github.event.inputs.commit }}
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.inputs.commit }}
```
Comment (tomhoule) [inline .github/workflows/build-wasm.yml:18]: same here

File: .github/workflows/build-wasm.yml (hunk 1 above)
Comment (pimeys) [reply .github/workflows/build-wasm.yml:13]: It doesn't even look like doing anything...

File: .github/workflows/build-wasm.yml (hunk 2 above)
Comment (pimeys) [reply .github/workflows/build-wasm.yml:18]: Try taking it out and see if it still works.

Comment (tomhoule) [reply .github/workflows/build-wasm.yml:15]: I took it out and the code isn't being checked out, so let's keep it in^

File: .github/workflows/build-wasm.yml (hunk 1 above)
Comment (tomhoule) [reply .github/workflows/build-wasm.yml]: This one indeed wasn't necessary :heavy_check_mark:

File: .github/workflows/build-wasm.yml (hunk 2 above)
Comment (janpio) [reply .github/workflows/build-wasm.yml:15]: Only using the `uses: actions/checkout@v2` line as a job here should be fine - you only need explicit input usage when you are doing a `workflow_dispatch` or something and not a `push` or `pull_request` event.

**Pull request merged**